source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+                           the Find panels
+                           and panels opened by plugins
+========================   ===============
+
+Additionally,
+the following four menus open
+when you click the respective section
+in the status bar:
+
+- **Encoding**
+- **Line Endings**
+- **Indentation**
+- **Syntax**
+
+.. figure:: images/statusbar_menu.gif
+   :class: shadowed
+
+   Demonstration of a status bar menu.
+
+
+.. _menu-items:
+
+Menu Items
+==========
+
+A menu item can either
+invoke a command (with arguments)
+when it is selected,
+or have a submenu.
+
+The available properties are:
+
+- a command name,
+- arguments for the command,
+- an ID,
+- a caption,
+- a mnemonic and
+- a submenu.
+
+In order to function properly,
+a menu item must provide at least:
+
+- a command name,
+- a caption and a submenu,
+- just a caption, or
+- an ID (see :ref:`below <item-ids>`).
+
+When parsing a menu item,
+the following rules apply:
+
+#. A menu item with a submenu
+   cannot invoke command.
+   It also overrides the :ref:`separator caption <menu-separators>`
+   and causes the item to render
+   a literal hyphen as caption.
+
+#. If no caption is provided,
+   a caption is inferred
+   from the command's ``description`` method.
+   If neither caption nor command are provided,
+   the caption will be an empty string.
+
+#. The character used for the mnemonic
+   must be contained in the item's caption.
+   Mnemonics are case-sensitive.
+
+#. Menu items referencing
+   commands that cannot be found
+   are disabled.
+
+#. Menu items can be hidden or disabled
+   by their referenced command.
+
+.. TODO add refs to Command methods
+
+
+.. _menu-separators:
+
+Separators
+**********
+
+Separators are menu items
+with the caption ``-``
+and no submenu.
+They are commonly used
+to group menu items with similar purposes
+or that are otherwise related.
+Separators cannot invoke commands.
+The presence of a submenu
+causes the menu item
+to be rendered as a regular item
+with a single hyphen as its caption.
+
+Multiple consecutive separators are reduced to one
+and separators at the beginning or the end of a menu
+are not displayed.
+
+
+.. _menu-merging:
+
+Menu Merging
+============
+
+``.sublime-menu`` files are loaded
+in the same order as packages
+Menu files with the same name are concatenated,
+unless IDs are specified
+(see :ref:`below <item-ids>`).
+
+Menu files in the same package
+are loaded in lexicographical order
+starting at the root folder,
+and then traversing sub-folders in the same manner.
+
+As a special case,
+menu items from the *User* package
+declared in the standard non-ID section
+are always inserted after any standard items
+from other packages.
+
+
+.. _item-ids:
+
+Item IDs
+********
+
+When a menu item specifies an ID,
+a separate section within the menu is searched for
+and, if it does not exist,
+created at the end of the menu.
+*This ID lookup is forward-going only.*
+
+The ID determines the section's name

```

Comment:
name,

otherwise it reads as "... determine this AND that ...", which I don't think it's the intention here.